evm-contracts/src/v0.7/ChainlinkClient.sol

```diff
@@ -264,16 +329,30 @@ contract ChainlinkClient {
       bytes memory
     )
   {
-    return abi.encodeWithSelector(
-      oracle.oracleRequest.selector,
-      SENDER_OVERRIDE, // Sender value - overridden by onTokenTransfer by the requesting contract's address
-      AMOUNT_OVERRIDE, // Amount value - overridden by onTokenTransfer by the actual amount of LINK sent
-      req.id,
-      req.callbackAddress,
-      req.callbackFunctionId,
-      req.nonce,
-      dataVersion,
-      req.buf.buf);
+    if (dataVersion == 2) {
```

Comment:
I'm not sure how good the Solidity optimizer is, maybe worth experimenting, but I think this method will take up a lot of extra unused bytecode by always keeping both code paths for both data versions around. I believe that by passing the data selector in as an argument with the data version then you can cut down the code path size. Alternatively you could just have a bytes4 variable defined by a ternary expression at the top of the method, and then use that as the first argument in the encoding. The thing is I think the encoding takes up a lot of space so having it twice in the method will take up a lot of bytecode space.

Reply:
Changed to use a ternary operator.

With regards to dropping multiword support in v0.7 in favour of v0.8: We have an upcoming story to move the Operator contract to v0.8. I think that's a more logical point to make this change.